### ISSUE TITLE
[BUG] Intermediate dropout and weight decay settings seem to have no effect

### ISSUE THREAD

**Issue by tmostak** [h2oai/h2o-llmstudio#432]
### 🐛 Bug

Intermediate dropout and weight decay settings seem to have no impact on models (at least for CodeLlama models)

### To Reproduce

I've tested almost all available hyperparameters for non-rlhf training, and all have observable impacts on the model (as evidenced by perplexity and other measures of downstream performance), except for intermediate dropout and weight decay. In the debugger, I can see intermediate dropout setting `hidden_dropout_prob` and `attention_probs_dropout_prop` in `update_backbone_config`, and I can see `weight_decay` being set in `get_optimizer`, however regardless of whether I set intermediate dropout to 0 or 0.5, and weight decay at 0 or 0.5, there is no change in the perplexity after model training.

### LLM Studio version
Master latest commit

**Comment by tmostak** [comment]
I should note that I'm doing a full fine tune and not using LORA.

**Comment by tmostak** [comment]
Thanks @maxjeblick, that makes more sense now. Is there any fundamental reason why the Llama model config doesn't expose dropout, or was this just an oversight that could be remedied? (Found this [issue](https://github.com/facebookresearch/llama/issues/231) but there has been no clear answer given yet)

Regarding `weight_decay`, yes I don't see any changes at all in the perplexity when changing it anywhere from 0 to very high values (i.e. 0.5), but easily see changes with even small changes of parameters like learning rate or gradient clip. Were you able to see changes in model validation metrics when changing `weight_decay`?

**Comment by maxjeblick** [comment]
I'll investigate the weight decay peculiarity these days.

W.r.t. dropout, I'm not sure what is the reason behind disabling this feature completely. I assume that llama was (pre-) trained without dropout, and the authors didn't bother adding it back.

**Comment by maxjeblick** [comment]
Sorry for the late reply!
I've checked training of `CodeLlama-7b-Python-hf` with default settings, once with weight decay 0, once with weight decay 0.5, and validation metrics are different.

![Screenshot 2023-10-17 at 09-56-37 H2O LLM Studio](https://github.com/h2oai/h2o-llmstudio/assets/24281881/f1a988ea-5efd-459d-ad84-1dd9682c598d)
![Screenshot 2023-10-17 at 09-58-52 H2O LLM Studio](https://github.com/h2oai/h2o-llmstudio/assets/24281881/4042fc64-2e4d-4336-8caf-e0cc438769cd)
![Screenshot 2023-10-17 at 09-58-42 H2O LLM Studio](https://github.com/h2oai/h2o-llmstudio/assets/24281881/0a7c0243-5a94-437a-b360-68fd220029f2)


Regarding the dropout, I'll create a PR that will handle this better.
